docs/_packages/socket_mode.md

```diff
@@ -0,0 +1,239 @@
+  });
+```
+---
+
+### Lifecycle events
+
+The client's connection to Slack has a lifecycle. This means the client can be seen as a state machine which transitions
+through a few states as it connects, disconnects, reconnects, and synchronizes with Slack. The client emits an event
+for each state it transitions to throughout its lifecycle. If your app simply needs to know whether the client is
+connected or not, the `.connected` boolean property can be checked.
+
+In the table below, the client's states are listed, which are also the names of the events you can use to observe
+the transition to that state. The table also includes description for the state, and arguments that a listener would
+receive.
+
+| Event Name      | Arguments | Description |
+|-----------------|-----------------|-------------|
+| `connecting`    |  | The client is in the process of connecting to the platform. |
+| `authenticated` | `(connectData)` - the response from `apps.connections.open` | The client has authenticated with the platform. This is a sub-state of `connecting`. |
+| `connected`     |  | The client is connected to the platform and incoming events will start being emitted. |
+| `ready`         |  | The client is ready to send outgoing messages. This is a sub-state of `connected` |
+| `disconnecting` |  | The client is no longer connected to the platform and cleaning up its resources. It will soon transition to `disconnected`. |
+| `reconnecting`  |  | The client is no longer connected to the platform and cleaning up its resources. It will soon transition to `connecting`. |
+| `disconnected`  | `(error)` | The client is not connected to the platform. This is a steady state - no attempt to connect is occurring. The `error` argument will be `undefined` when the client initiated the disconnect (normal). |
+
+The client also emits events that are part of its lifecycle, but aren't states. Instead, they represent specific
+moments that might be interesting to your app. The following table lists these events, their description, and includes
+the arguments that a listener would receive.
+
+| Event Name      | Arguments | Description |
+|-----------------|-----------|-------------|
+| `error`         | `(error)` | An error has occurred. See [error handling](#handle-errors) for details. |
+| `slack_event`   | `(eventType, event)` | An incoming Slack event has been received. |
+| `unable_to_socket_mode_start` | `(error)` | A problem occurred while connecting, a reconnect may or may not occur. |
+
+---
+
+### Logging
+
+The `SocketModeClient` will log interesting information to the console by default. You can use the `logLevel` to decide how
+much information, or how interesting the information needs to be, in order for it to be output. There are a few possible
+log levels, which you can find in the `LogLevel` export. By default, the value is set to `LogLevel.INFO`. While you're
+in development, its sometimes helpful to set this to the most verbose: `LogLevel.DEBUG`.
```

Comment:
```suggestion
in development, it's sometimes helpful to set this to the most verbose: `LogLevel.DEBUG`.
```

Reply:
Have we thought about turning on `LogLevel.DEBUG` when in development? For example, when there is no set value for `process.env.NODE_ENV`?

I think this would be helpful to developers and also issues submitted to the Issue tracker.